frontend/src/component/operator-wizard/element-component/field-factory/DropdownField.js

```diff
@@ -155,11 +181,97 @@ export class DropdownField extends Component {
 
   onSelect = (_, event) => {
     let value = event.target.value;
+    const {
+      currentSteps,
+      originalSteps,
+      currentPages,
+      originalPages
+    } = this.props;
+    var copyOfCurrentSteps = [],
+      copyOfCurrentPages = [],
+      copyOfOriginalPages = Formatter.deepCloneArrayOfObject(originalPages);
 
     this.isValidField(value);
     this.reBuildChildren(value);
 
     this.props.props.page.loadPageChildren();
+
+    if (Validator.isEmptyArray(currentSteps)) {
+      copyOfCurrentSteps = Formatter.deepCloneArrayOfObject(originalSteps);
+    } else {
+      copyOfCurrentSteps = Formatter.deepCloneArrayOfObject(currentSteps);
+    }
+
+    if (Validator.isEmptyArray(currentPages)) {
+      copyOfCurrentPages = Formatter.deepCloneArrayOfObject(originalPages);
+    } else {
+      copyOfCurrentPages = Formatter.deepCloneArrayOfObject(currentPages);
+    }
+
+    if (value.indexOf("rhdm") > -1) {
+      // remove the Smart Router item from steps
+      copyOfCurrentSteps = Formatter.filter(copyOfCurrentSteps, step => {
+        let subStepList = [];
+        if (step.name === "Components") {
+          subStepList = Formatter.filter(step.steps, subStep => {
+            return subStep.name !== "Smart Router";
+          });
+          step.steps = subStepList;
+          return step;
+        } else {
+          return step;
+        }
+      });
+
+      // remove the Smart Router item from pages
+      copyOfCurrentPages = Formatter.filter(copyOfCurrentPages, page => {
+        let subPageList = [];
+        if (page.label === "Components") {
+          subPageList = Formatter.filter(page.subPages, subPage => {
+            return subPage.label !== "Smart Router";
+          });
+          page.subPages = subPageList;
+          return page;
+        } else {
+          return page;
+        }
+      });
+
+      //update the value of the related field.
```

Comment:
From line 240 up to 274 looks like code that is duplicated. Can this be refactored into a new method or take out of the if/else the duplicated code?